### DISCUSSION TITLE
Is there a way to calculate animation duration?

### DISCUSSION THREAD

**Discussion by msaspence** [pmndrs/react-spring#1199] [Support]
Is there a way to get how long the animation will take to complete?

I'm trying not to revert to using the `duration` configuration but wish to know so that I can set a timeout to trigger other code once the animation is done.

**Comment by DanielBalog86** [comment]
Could you not use the `onRest` callback? This is a callback that is part of the common props that gets triggered when the animation finishes. 

You also have an `onStart` callback that triggers when the animation starts. So by combining both you could have something happen while the animation is running, regardless of duration.

**Reply by msaspence** [reply]
I'd had problems with using `onRest` because of the combinations of components I'm using.

I'm struggling to describe why I can't use `onRest` but maybe my example will describe it better...

https://gist.github.com/msaspence/4f14ba4e41108ec46f77731e713cec8e

This also demonstrates the solution I currently have, which works. I'm not totally sure I've explored the performance consequences of this. The promise is going to be different each render, but maybe DelayedUnmount is light enough for that not to matter.

I'm sure there is a better way to achieve this with out having to resort to this promise "cleverness" (read hack). But I don't see it without coupling `DelayedUnmount` to the animation.
